frontend/src/metabase/admin/performance/components/StrategyForm.tsx

```diff
@@ -0,0 +1,271 @@
+import { useFormikContext } from "formik";
+import type { ReactNode } from "react";
+import { useEffect } from "react";
+import { t } from "ttag";
+import _ from "underscore";
+
+import type { FormTextInputProps } from "metabase/forms";
+import {
+  Form,
+  FormProvider,
+  FormRadioGroup,
+  FormSubmitButton,
+  FormTextInput,
+  useFormContext,
+} from "metabase/forms";
+import { color } from "metabase/lib/colors";
+import {
+  Button,
+  Group,
+  Icon,
+  Loader,
+  Radio,
+  Stack,
+  Text,
+  Title,
+} from "metabase/ui";
+
+import { rootId } from "../constants";
+import { useRecentlyTrue } from "../hooks/useRecentlyTrue";
+import type { Strategy, StrategyType } from "../types";
+import { Strategies } from "../types";
+import { strategyValidationSchema } from "../validation";
+
+export const StrategyForm = ({
+  targetId,
+  setIsDirty,
+  saveStrategy,
+  savedStrategy,
+}: {
+  targetId: number | null;
+  setIsDirty: (isDirty: boolean) => void;
+  saveStrategy: (values: Strategy) => Promise<void>;
+  savedStrategy?: Strategy;
+}) => {
+  return (
+    <FormProvider<Strategy>
+      key={targetId}
+      initialValues={savedStrategy ?? { type: "inherit" }}
+      validationSchema={strategyValidationSchema}
+      onSubmit={saveStrategy}
+      enableReinitialize={true}
+    >
+      <StrategyFormBody targetId={targetId} setIsDirty={setIsDirty} />
+    </FormProvider>
+  );
+};
+
+const StrategyFormBody = ({
+  targetId,
+  setIsDirty,
+}: {
+  targetId: number | null;
+  setIsDirty: (isDirty: boolean) => void;
+}) => {
+  const { dirty, values, setFieldValue } = useFormikContext<Strategy>();
+
+  const selectedStrategyType = values.type;
+
+  useEffect(() => {
+    setIsDirty(dirty);
+  }, [dirty, setIsDirty]);
+
+  useEffect(() => {
+    if (selectedStrategyType === "duration") {
+      setFieldValue("unit", "hours");
+    }
```

Comment:
enums for all the things! avoid any hard coded strings. future you will appreciate it when you need to refactor this feature and you're trying to search for every instance of the word "unit", "duration", or "hours" in the codebase.